components/wifi/wifi_ap.c

```diff
@@ -24,63 +31,110 @@ static void wifi_event_handler(void* arg, esp_event_base_t event_base,
     }
 }
 
-esp_err_t wifi_init_softap(void)
+static esp_err_t start_softap(const char* ssid, const char* passwd)
 {
-    esp_err_t ret = esp_netif_init();
-    if (ret != ESP_OK) {
-        return ret;
-    }
-    
-    ret = esp_event_loop_create_default();
-    if (ret != ESP_OK) {
-        return ret;
-    }
-
-    esp_netif_create_default_wifi_ap();
-
-    wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
-    ret = esp_wifi_init(&cfg);
-    if (ret != ESP_OK) {
-        return ret;
-    }
-
-    ret = esp_event_handler_instance_register(WIFI_EVENT,
-                                              ESP_EVENT_ANY_ID,
-                                              &wifi_event_handler,
-                                              NULL,
-                                              NULL);
-    if (ret != ESP_OK) {
-        return ret;
-    }
-
+    esp_err_t ret = ESP_OK;
     wifi_config_t wifi_config = {
         .ap = {
-            .ssid = EXAMPLE_ESP_WIFI_SSID,
-            .ssid_len = strlen(EXAMPLE_ESP_WIFI_SSID),
+            .ssid = "",
+            .ssid_len = 0,
             .channel = EXAMPLE_ESP_WIFI_CHANNEL,
-            .password = EXAMPLE_ESP_WIFI_PASS,
+            .password = "",
             .max_connection = EXAMPLE_MAX_STA_CONN,
             .authmode = WIFI_AUTH_WPA_WPA2_PSK
         },
     };
-    if (strlen(EXAMPLE_ESP_WIFI_PASS) == 0) {
+
+    memcpy(wifi_config.ap.ssid, ssid, MIN(strlen(ssid), sizeof(wifi_config.ap.ssid)));
+    wifi_config.ap.ssid_len = MIN(strlen(ssid), sizeof(wifi_config.ap.ssid));
+    memcpy(wifi_config.ap.password, passwd, MIN(strlen(passwd), sizeof(wifi_config.ap.password)));
```

Comment:
Because passwd don't has a matching length field, I think you need to copy '\0', too. So the length is not right. And it seems it is better to use strncpy here.

Reply:
Accepted